### PR TITLE
nl: move from getopts to clap

### PR DESCRIPTION
**Pull request by LevitatingBusinessMan**
For https://github.com/uutils/coreutils/issues/1864
There shouldn't be any behavioral changes.

### CODE REVIEW

File: src/uu/nl/src/nl.rs [hunk 1]
```diff
@@ -67,78 +68,108 @@ enum NumberFormat {
...
+            .long(options::HEADER_NUMBERING)
+            .help("use STYLE for numbering header lines")
+            .value_name("STYLE")
+        )
+        .arg(
+            Arg::with_name(options::LINE_INCREMENT)
+            .short("i")
+            .long(options::LINE_INCREMENT)
+            .help("line number increment at each line")
+            .value_name("")
```
Comment (ycd) [inline src/uu/nl/src/nl.rs:127]: I see that the existing implementation has empty value but GNU implementation has `NUMBER` for value name could you also add it?

```rust
       -i, --line-increment=NUMBER
              line number increment at each line
```

**Review by ycd — CHANGES_REQUESTED** [review]
please run `rustfmt` to your code.

**Review by ycd — APPROVED** [review]
looks great, thanks!

**Comment by sylvestre** [comment]
Indeed, thanks :)

**Pull request merged**